Feature/kjb/places service

KlaasH opened the pull request:
The comment that I wrote then apparently accidentally threw away on https://github.com/azavea/pfb-network-connectivity/pull/347 was to the effect that the `getPlace` function was very repetitive of the one in the places detail controller.

Because I felt a little bad about requesting a change of that scale in the first place, and even worse about the thought of not requesting initially and then coming back with it on a second round, and to prove to myself that what I was envisioning wasn't crazy, I've made this sub-PR.  It should leave both the compare and place detail views working the same, but with the `getPlaces` function abstracted and shared.

flibbertigibbet commented on src/angularjs/src/app/places/detail/places-detail.controller.js:
```diff
@@ -25,53 +25,30 @@
-                if (!data.results || !data.results.length) {
-                    $log.warn('no matching analysis job found for neighborhood ' + uuid);
+            ctl.mapLayers = {};
+            Places.getPlace(uuid).then(function (place) {
```
Might be worth adding some error handling for this promise (pass a second function). There are some handled cases for results not being found that will resolve successfully with an empty object, but in case of unhandled errors such as the query failing altogether it might be good to throw in a case for that here, even if it's just a log message and nulling out variables.

KlaasH commented:
Pushed a commit to log an error and clear the controller properties on failure.  If this looks good, go ahead and pull it in however you like (as is, squashed, squashed-and-tweaked) and with this and the `ui-sref` fix https://github.com/azavea/pfb-network-connectivity/pull/347 should be good to go.

flibbertigibbet commented:
Thanks for this refactor.
